text/0007-gas-price-adjustment.md

```diff
@@ -0,0 +1,134 @@
+
+## Guide-level explanation
+
+[guide-level-explanation]: #guide-level-explanation
+
+The algorithm sets a `BASE_PRICE` for every block, which the included transactions are charged by default. The paying signer accepts to be charged minimum `BASE_PRICE`, and must also specify a `MAX_GAS_PRICE` they are willing to accept as well. On top of `BASE_PRICE`, each transaction can specify a `PRICE_PREMIUM` which is added to `BASE_PRICE` to calculate the final gas price the transaction is charged. This is to enable users to prioritize their transactions if a congestion happens in spite of price adjustment.
+
+`BASE_PRICE` starts at `INITIAL_BASE_PRICE` at genesis, remains constant for `ADJUSTMENT_PERIOD` ticks. Once that period is over, it can be updated once, in a system transaction. The process repeats at the end of each consecutive period of the same length.
+
+The update can be either a percent increase or decrease of `BASE_PRICE`, specified by an `ADJUSTMENT_RATE`. To decide on the direction of the change, we make use of the fullnesses of blocks from the preceding adjustment period. We map the array of fullnesses to a single value called `FULLNESS` (details on how to compute this are given below). Then, if `FULLNESS` is greater than `TARGET_FULLNESS`, we increase the price. Otherwise, we decrease the price.
+
+## Reference-level explanation
+
+[reference-level-explanation]: #reference-level-explanation
+
+### Protocol Variables
+
+- `BASE_PRICE`: The value of the price floor, in motes. Given blocks proposed in a certain adjustment period, all transactions must be charged at least this value. The value will be included in each block in a field of the same name.
+- `FULLNESS`: A value representing the fullness of blocks of a given adjustment period.
+
+### Protocol Parameters
+
+- `INITIAL_BASE_PRICE`: The value of `BASE_PRICE` at genesis. Also the *minimum value* for `BASE_PRICE`. Value: TBD.
+- `ADJUSTMENT_RATE`: The rate at which `BASE_PRICE` can move up or down per day. Value: `10_000_000_000` (1%).
+- `DENOMINATOR`: Fractions will be represented as trillionths of 1. Value: `1_000_000_000_000`.
+- `BLOCK_GAS_LIMIT`: Maximum gas allowed in a block. Value: TBD.
+- `TARGET_FULLNESS`: The value which will determine whether `BASE_PRICE` should move up or down. Value: `650_000_000_000` (65%).
+- `MIN_FINALIZED_BLOCKS`: The minimum number of blocks that must be finalized in an adjustment period. Otherwise, a fail-safe is triggered. Value: TBD.
+
+### Transaction Fields
+
+- `PRICE_PREMIUM`: The price premium specified for a given transaction.
```

Comment:
Yeah, so, this is the part that I don't like from EIP-1559 and was, originally, thinking that we dropped it. The problem with a day-by-day plodding price discovery is that, _behaviorally_, it makes not an iota of difference to the users whether their money is going to the validators or poofing into non-existence. If the base price isn't matching what the people who, on the Ethereum platform, would have paid while generating a price peak, _we will still get the same price peak_. It's just that some of the tokens may be burnt (actually, are we proposing to do that, or not? Can't find it in the CEP, and that should be addressed). So we will have constructed this whole complicated machine for nothing.

My strong preference is for getting rid of bids altogether, making the price adjust quicker (every few eras perhaps), and hoping that we can implement genuine gas reservation before someone makes the first usable accelerator (given low use for the first two year, I think this is a realistic expectation).